kotlin/workflow-runtime/src/main/java/com/squareup/workflow/internal/Workers.kt

```diff
@@ -46,16 +47,35 @@ internal fun <T> CoroutineScope.launchWorker(
     .transformToValueOrDone()
     .catch { e ->
       // Workers that failed (as opposed to just cancelled) should have their failure reason
-      // re-thrown from the workflow runtime. If we don't unwrap the cause here, they'll just
-      // cause the runtime to cancel.
-      val cancellationCause = e.unwrapCancellationCause()
-      throw cancellationCause ?: e
+      // re-thrown from the workflow runtime.
+      throw e.wrapWorkerLaunchException(worker::class)
```

Comment:
The `worker.run()` call on line 44 is the one that we should wrap. Doing it here is confusing and doesn't communicate clearly the actual issue we're trying to handle.

Perhaps define a private extension like `worker.runWithNullCheck()` that can do the try/catch.

Reply:
Looks like the crash actually happens within `transformToValueOrDone()`, where we `collect` from the Flow

Reply:
I thought the crash was caused by `run` returning null though - is that not true?

Reply:
Indeed - but it seems as though the crash isn't actually thrown until `collect` is used. The extension function on Flow must be causing the crash in the generated static method.

Reply:
It doesn't really matter where the crash currently happens, the point here is to try to catch the invalid behavior as soon as possible. The cancellation unwrap code is weird and already confusing, and adding more logic there, especially logic that is actually about a different part of the code altogether, just compounds the confusion. 

Checking for the null where it actually happens makes the intent of this code much clearer.

Reply:
I've implemented what I suggested and pushed this branch. Let me know what you think.